models/host.tsp

```diff
@@ -16,13 +17,24 @@ namespace rpp.host {
         };
     }
 
+    model HostAttr {
+        ...HostInfoAttr;
+        ...rpp.dns.DNS
+    }
+
     model NSHostObj {
-        hostObj: HostInfoObj[];
+        hostObj: HostInfoObj[] | DNSHostObj[];
```

Comment:
That's still referenced in rpp.domains.NS, that is used in rpp.domains.Domain.
Eventually HostObject reference in Domains object now has dns structure.

I think Domains needs an own model for HostObject reference (like NSHostAttr already is) so this model does not have this spillover effect

Reply:
There seems to be no good way to have two mutual exclusive models. For now let's start with an optional model.